### ISSUE TITLE
We still need manual registeration of nodes alongisde the more flexible method based on plugins

### ISSUE THREAD

**Issue by facontidavide** [MarqRazz/ActionServer.BT#4]
I have usersthat still prefer not to use plugins, but static linking.

I suggest adding a method like this:

```c++
  template <typename T, typename... Args> void registerNodeType(const std::string& ID, ExtraArgs... args)
```
  
  Similar to: https://github.com/BehaviorTree/BehaviorTree.CPP/blob/master/include/behaviortree_cpp/bt_factory.h#L357-L358
  
  Or we can just expose the inner factory in the public API:
  
```c++
 BT:: BehaviorTreeFactory& ActionServerBT::factory();
```

Thoughts?

**Comment by MarqRazz** [comment]
I like the approach of exposing the `factory` better. Should we add a callback the user can define that way [if new behavior plugin's are loaded at runtime](https://github.com/MarqRazz/ActionServer.BT/blob/main/action_server_bt/src/action_server_bt.cpp#L88-L93) we will also reload the statically defined ones? (I clear the factory on each reload [here](https://github.com/MarqRazz/ActionServer.BT/blob/0f6162c0222e780f53eb64b4014723ecf0106c06/action_server_bt/src/bt_utils.cpp#L133))

**Comment by facontidavide** [comment]
> if new behavior plugin's are loaded at runtime we will also reload the statically defined ones?

Yes, that might be a problem, because of the cleanup.

This could be solve with... a callback? But as the number of potential callbacks increases, I start thinking that maybe inheritance with virtual methods to iplement these "customization points" might be better.

**Comment by facontidavide** [comment]
solved by #6
